### PR TITLE
Checking for Anagram

### PR DESCRIPTION
**Pull request by prateek-jpg**
Have solved the issue in java. Please let me know if any changes required.
#395 fixed

### CODE REVIEW

**Comment by prateek-jpg** [comment]
@tarun26091999 Have solved the issue.

File: Code/Java/Anagram.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
+
+/*Two strings are said to be an Anagram if all the characters of a string are equal and have same frequency but arranged in different order.
```
Comment (raksha009) [inline Code/Java/Anagram.java:2]: Write /*  
at line 1

File: Code/Java/Anagram.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+import java.util.*;
+public class Anagram {
+
+	public static void main(String[] args) {
+	
+		Scanner sc=new Scanner(System.in);
+		
+		System.out.println("Enter string 1:");
+		String s=sc.next();
```
Comment (raksha009) [inline Code/Java/Anagram.java:16]: Add spaces around the operators like =, < , > etc

File: Code/Java/Anagram.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+		System.out.println("Enter string 2:");
+		String s1=sc.next();
```
Comment (raksha009) [inline Code/Java/Anagram.java:18]:
```suggestion
		String s1 = sc.next();
```

File: Code/Java/Anagram.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+		        
+		int l1=s.length(),l2=s1.length(); // storing the length of both the strings
```
Comment (raksha009) [inline Code/Java/Anagram.java:20]: Add comments above the line

File: Code/Java/Anagram.java [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+			
+			s="";s1="";
+			Arrays.sort(c1);Arrays.sort(c2);//sorting both the array.
+			
+			for(char ch:c1) {
+				s=s+ch;
+			}
+			
+			for(char ch:c2) {
+				s1=s1+ch;
```
Comment (raksha009) [inline Code/Java/Anagram.java:36]:
```suggestion
				s1=s1 + ch;
```

File: Code/Java/Anagram.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    1. Enter string 1-prateek
+	   Enter string 2-eektarp
+       OUTPUT-Two strings are Anagram
+    
+    2. Enter string 1-abcde
+       Enter string 2-cdba       
+      OUTPUT-Two strings are not Anagram
+      
+      
+      Time Complexity-O(nlog n)
```
Comment (raksha009) [inline Code/Java/Anagram.java:58]: Specify n

File: Code/Java/Anagram.java [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+				System.out.println("OUTPUT:Two strings are Anagram");
+			}else
+				System.out.println("OUTPUT:Two strings are not Anagram");
+				
+		}
+	}
+}
+
+
+/*  OUTPUT
```
Comment (raksha009) [inline Code/Java/Anagram.java:48]:
```suggestion
/*  
Test Cases
```

File: Code/Java/Anagram.java [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+    1. Enter string 1-prateek
```
Comment (raksha009) [inline Code/Java/Anagram.java:49]:
```suggestion
    Input:  Enter string 1-prateek
```

File: Code/Java/Anagram.java [hunk 9]
```diff
@@ -0,0 +1,61 @@
...
+	   Enter string 2-eektarp
+       OUTPUT-Two strings are Anagram
```
Comment (raksha009) [inline Code/Java/Anagram.java:51]:
```suggestion
       Output:  Two strings are Anagram
```

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/Anagram.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+		System.out.println("Enter string 1: ");
+		String s = sc.next();
+		System.out.println("Enter string 2: ");
+		String s1 = sc.next();
+		 // storing the length of both the strings       
+		int l1 = s.length(),l2 = s1.length(); 
+		
+		if(l1!=l2) {
+			//if both are of unequal length then its for sure that they are not an anagram
+			System.out.println("OUTPUT:Two strings are not Anagram");
```
Comment (raksha009) [inline Code/Java/Anagram.java:24]:
```suggestion
			System.out.println("Two strings are not Anagram"); 
```

File: Code/Java/Anagram.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+			
+			for(char ch:c1) {
+				s = s + ch;
+			}
+			
+			for(char ch:c2) {
+				s1 = s1 + ch;
+			}
+			if(s.compareTo(s1)==0) {
+				System.out.println("OUTPUT:Two strings are Anagram");
```
Comment (raksha009) [inline Code/Java/Anagram.java:41]: Hey, we do not have to display "OUTPUT" on the console, its just a part of documentation when we add test cases after our code in the file.

File: Code/Java/Anagram.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+			}else
+				System.out.println("OUTPUT:Two strings are not Anagram");
```
Comment (raksha009) [inline Code/Java/Anagram.java:43]:
```suggestion
				System.out.println("Two strings are not Anagram");
```

File: Code/Java/Anagram.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+				
+		}
+	}
+}
+
+
+/*  
+             OUTPUT
```
Comment (raksha009) [inline Code/Java/Anagram.java:51]: Test cases
Input: 
Output: 

Also plz align properly.

File: Code/Java/Anagram.java (hunk 6 above)
Comment (raksha009) [reply Code/Java/Anagram.java:61]: Do this chnge too.

File: Code/Java/Anagram.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+       OUTPUT:  Two strings are Anagram
+    
+    INPUT: Enter string 1: abcde
+           Enter string 2: cdba       
+      OUTPUT:  Two strings are not Anagram
+      
+      
+      Time Complexity-O(nlog n)
+      Space Complexity-O(n) 
+      where n: size of the input
```
Comment (raksha009) [inline Code/Java/Anagram.java:63]: n cannot be size of input. Its is length of / no of our something depending on our problem statement

**Review by raksha009 — CHANGES_REQUESTED** [review]

File: Code/Java/Anagram.java (hunk 13 above)
Comment (raksha009) [reply Code/Java/Anagram.java:51]: Test cases
Input:
Output:

Also plz align properly.

Make this change

File: string/readme.md [hunk 15]
```diff
@@ -15,4 +15,12 @@ A string is a class that contains a char array, but automatically manages it for
 
 * Convert string to lowercase and uppercase ---->[C++](/Code/C++/string_to_lowercase_and_uppercase.cpp)
 * Revere a string using Stack ---->[C++](/Code/C++/reverse_a_string_using_stack.cpp)
+
+
+* Checking for Anagram ---->[Java](/Code/Java/Anagram.java)
+
+* Reverse a string ---->[C++](/Code/C++/reverse_string.cpp)
```
Comment (raksha009) [inline string/readme.md:22]: Plz have a look of what changes have you done via Files changed option and make changes appropriately.

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by rudrakshi99** [comment]
PLease resolve merge conflicts.

**Review by raksha009 — APPROVED** [review]

**Comment by raksha009** [comment]
There are conflicts that can't be resolved by us but only participants.
This being the case, closing the PR as no more commits are allowed.

**Pull request closed without merging**